x-pack/plugins/session_view/public/components/Process/index.tsx

```diff
@@ -0,0 +1,286 @@
+      margin-left: 8px;
+    `;
+
+    const buttons = [];
+
+    if (!isSessionLeader && process.children.length > 0) {
+      const childrenExpandedIcon = childrenExpanded ? 'arrowUp' : 'arrowDown';
+      const iconCSS = `margin-left: 8px;`;
+
+      buttons.push(
+        <EuiButton css={buttonCSS} onClick={() => setChildrenExpanded(!childrenExpanded)}>
+          <FormattedMessage id="kbn.sessionView.childProcesses" defaultMessage="Child processes" />
+          <EuiIcon css={iconCSS} size="s" type={childrenExpandedIcon} />
+        </EuiButton>
+      );
+    }
+
+    return buttons;
+  }
+
+  /**
+   * gets border, bg and hover colors for a process
+   */
+  function getProcessColors() {
+    let bgColor = 'none';
+    let hoverColor = '#6B5FC6';
+    let borderColor = 'transparent';
+
+    // if (props.isSummaryMatch) {
+    //   borderColor = '#8070F1'
+    //   bgColor = '#6B5FC629'
+    // }
+
+    // if (props.highlightAlert) {
+    //   bgColor = props.alertLevel > 0 ? props.theme['--color-red-500-16'] : props.theme['--color-cmdyellow-16']
+    // }
+
+    // if (props.alertLevel >= 0) {
+    //   if (props.alertLevel > 0) {
+    //     borderColor = props.theme['--color-red-500']
+    //     hoverColor = props.theme['--color-red-500-16']
+    //   } else {
+    //     borderColor = props.theme['--color-cmdyellow']
+    //     hoverColor = props.theme['--color-cmdyellow-16']
+    //   }
+    // }
+
+    return { bgColor, borderColor, hoverColor };
+  }
+
+  const { bgColor, borderColor, hoverColor } = getProcessColors();
+
+  const processCSS = `
+    position: relative;
+    display: block;
+    cursor: pointer;
+
+    &:not(:first-child) {
+      margin-top: 8px;
+    }
+
+    &:hover:before {
+      opacity: 0.24;
+      background-color: ${hoverColor};
+    }
+
+    &:before {
+      position: absolute;
+      height: 100%;
+      pointer-events: none;
+      content: '';
+      margin-left: -${depth * TREE_INDENT}px;
+      border-left: 4px solid ${borderColor};
+      background-color: ${bgColor};
+      width: calc(100% + ${depth * TREE_INDENT}px);
+    }
+  `;
+
+  const wrapperCSS = `
+    position: relative;
+    padding-left: 8px;
+    vertical-align: middle;
+    color: ${euiTheme.colors.mediumShade};
+    word-break: break-all;
+    min-height: 24px;
+    line-height: 24px;
+  `;
+
+  function renderSessionLeader() {
+    const event = process.getLatest();
+    const { name, user } = event.process;
+    const sessionIcon = interactive ? 'consoleApp' : 'compute';
+
+    return (
+      <>
+        <EuiIcon type={sessionIcon} /> <b css={darkTextCSS}>{name}</b>
+        &nbsp;
+        <FormattedMessage id="kbn.sessionView.startedBy" defaultMessage="started by" />
+        &nbsp;
+        <EuiIcon type="user" /> <b css={darkTextCSS}>{user.name}</b>
+      </>
+    );
+  }
+
+  // TODO: not customizable for now (cmd previously offered a template string to render)
+  const template = (process: IProcess) => {
```

Comment:
just curious, will we keep with the ability to use custom templates in Kibana?

Reply:
Which templates are those? The template I'm referring to here is a feature in the existing cmd terminal view: 
![image](https://user-images.githubusercontent.com/16198204/139499252-f74dc25f-023c-4be5-a4bf-78b14724b554.png)

I'm hesitant to re-implement this as it's a bit risky to allow a custom template like this due to xss.

Reply:
Maybe just a list of prefab templates e.g simple, detailed, etc... That way we can keep the implementation as regular <react> components to avoid xss. Or even just a simple list of properties to render would be better than this template string.